js/fcoin.js

```diff
@@ -341,7 +341,7 @@ module.exports = class fcoin extends Exchange {
         let orderType = type;
         let request = {
             'symbol': this.marketId (symbol),
-            'amount': this.amountToPrecision (symbol, amount),
+            'amount': this.amountToString (symbol, amount),
```

Comment:
Actually, with the most recent version of CCXT, amountToPrecision always returns strings, so it should work as it was, or is there a particular issue with this amount in your case? Have you tried with the most recent version of CCXT?

```
fcoin.createOrder (ETH/BTC, limit, buy, 1, 1)
fetch:
 fcoin POST https://api.fcoin.com/v2/orders
Request:
 { 'User-Agent': 'Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.71 Safari/537.36',
  'FC-ACCESS-KEY': 'f1aceafa3a8a493880b14c6f23bc05e2',
  'FC-ACCESS-SIGNATURE': 'y8bn264EQmVi+wRABj6M5aha8HQ=',
  'FC-ACCESS-TIMESTAMP': '1536731318044',
  'Content-Type': 'application/json' }
 {"amount":"1","price":"1","side":"buy","symbol":"ethbtc","type":"limit"}
          string ↑   string ↑
```

Can you show your `verbose` request ?

Reply:
You are right, the update in Sep 7th already solved this issue. I should have update local copy more frequently. Just out of curiosity, what is the difference between string and number when calculating signature?

Reply:
> Just out of curiosity, what is the difference between string and number when calculating signature?

This depends on the exchange, but for fcoin, in particular, the signature is calculated from a json-encoded object, so, depending on the type of that value when it's json-encoded, it is either surrounded with double quotes (string) or left without quotes (number) – that yields two different strings from which the signature is calculated. Thus it yields two different signatures, because the two input strings differ. And only one of the two that is accepted by the exchange, that is the one with values as strings. Hope this answers your question:

```JavaScript
// pseudocode

json1 = json ({ "a": 1 })   // == "{\"a\":1}"
json2 = json ({ "a": "1" }) // == "{\"a\":\"1\"}"
json1 !== json2 // true
calculateSignature (json1) !== calculateSignature (json2) // true
```

Reply:
I'd close this PR for now, if you don't mind, but we'll be happy to see more code and contributions from you in future! ) Thx!

Reply:
yes, that answers my questions. thanks. p.s. Fcoin is so funny, using string for px and amt.